warehouse/query-core/src/main/java/datawave/query/index/lookup/RangeStream.java

```diff
@@ -250,8 +257,12 @@ public Iterator<QueryPlan> iterator() {
 
                 if (log.isDebugEnabled()) {
                     log.debug("Query returned a stream with a context of " + this.context);
+                    int count = 0;
                     for (String line : StringUtils.split(queryStream.getContextDebug(), '\n')) {
                         log.debug(line);
+                        if (maxLinesToPrint > 0 && ++count > maxLinesToPrint) {
```

Comment:
Consider adding a final log line that indicates that the output was truncated because it hit this max. This way we can differentiate between cases where the log didn't have what we expected and the log was truncated.

Reply:
good call, I'll add that in the next commit